symengine/expression.h

```diff
@@ -32,16 +32,16 @@ class Expression
     //! Construct Expression from `int`
     Expression(int n) : m_basic(integer(n)) {}
     //! Construct Expression from Basic
-    template <typename T, typename = typename std::enable_if<std::is_constructible<RCP<const Basic>, T &&>::value>::type>
+    template <typename T>
```

Comment:
It depends on how you want to be able to construct the expression. The original intent was to just enable the constructor if the internal ``RCP<const Basic>`` is constructible from the input argument (forwarding the construction, in a sense). Basically, the goal was to be able to centralise the construction in one place and to allow to write expressions in a "natural way".

What version of ICC is this? I think I remember it supporting `` = default`` and ``is_constructible`` in version 15.